engine/qclib/qcc_pr_comp.c

```diff
@@ -3834,7 +3839,11 @@ QCC_sref_t QCC_PR_StatementFlags ( QCC_opcode_t *op, QCC_sref_t var_a, QCC_sref_
 			TypeName(var_b.cast, typeb, sizeof(typeb));
 			QCC_PR_ParseWarning(WARN_STRICTTYPEMISMATCH, "'%s' type mismatch: %s with %s", op->name, typea, typeb);
 		}
-		if ((var_a.sym->constant && var_b.sym->constant && !var_a.sym->temp && !var_b.sym->temp) || (var_a.sym == var_b.sym && var_a.ofs == var_b.ofs))
+
+		pbool sym_cmp = var_a.sym == var_b.sym && var_a.ofs == var_b.ofs;
+		pbool nan_cmp = nan_eq_cond && (var_a.cast == type_float || var_a.cast == type_vector);
```

Comment:
Hold up, I don't need to check the types here.  What I get for writing code at 4 in the morning.